src/test/java/uk/gov/companieshouse/api/testdata/service/impl/CompanyProfileServiceImplTest.java

```diff
@@ -47,56 +59,149 @@ class CompanyProfileServiceImplTest {
         spec.setCompanyNumber(COMPANY_NUMBER);
+        savedProfile = new CompanyProfile();
+    }
+
+    // Test that a company profile is created with default company type with England and Wales jurisdiction
+    @Test
+    void createCompanyWithoutCompanyTypeAndWithEnglandWales() {
         spec.setJurisdiction(Jurisdiction.ENGLAND_WALES);
+        spec.setCompanyStatus(COMPANY_STATUS_ADMINISTRATION);
+        assertCompanyProfile( COMPANY_STATUS_ADMINISTRATION, Jurisdiction.ENGLAND_WALES.toString(), COMPANY_TYPE_LTD, false);
+    }
 
-        CompanyProfile savedProfile = new CompanyProfile();
+    // Test that a company profile is created with default company status with SCOTLAND jurisdiction
+    @Test
+    void createCompanyWithoutCompanyStatusAndWithScotland() {
+        spec.setJurisdiction(Jurisdiction.SCOTLAND);
+        spec.setCompanyType(COMPANY_TYPE_LTD);
+        assertCompanyProfile( COMPANY_STATUS_ACTIVE, Jurisdiction.SCOTLAND.toString(), COMPANY_TYPE_LTD, false);
+    }
+
+    // Test that a company profile is deleted
+    @Test
+    void delete() {
+        when(repository.findByCompanyNumber(COMPANY_NUMBER))
+                .thenReturn(Optional.of(savedProfile));
+
+        assertTrue(this.companyProfileService.delete(COMPANY_NUMBER));
+        verify(repository).delete(savedProfile);
+    }
+
+    // Test that a company profile is not deleted when it does not exist
+    @Test
+    void deleteNoCompanyProfile() {
+        when(repository.findByCompanyNumber(COMPANY_NUMBER))
+                .thenReturn(Optional.empty());
+
+        assertFalse(this.companyProfileService.delete(COMPANY_NUMBER));
+        verify(repository, never()).delete(any());
+    }
+
+    // Test that a company profile is created with dissolved company status
+    @Test
+    void createDissolvedCompany() {
+        spec.setJurisdiction(Jurisdiction.ENGLAND_WALES);
+        spec.setCompanyStatus(COMPANY_STATUS_DISSOLVED);
+        assertCompanyProfile( COMPANY_STATUS_DISSOLVED, Jurisdiction.ENGLAND_WALES.toString(), COMPANY_TYPE_LTD, true);
+    }
+
+    // Test that a company profile is created with plc company type
+    @Test
+    void createPlcCompany() {
+        spec.setJurisdiction(Jurisdiction.ENGLAND_WALES);
+        spec.setCompanyType(COMPANY_TYPE_PLC);
+        assertCompanyProfile( COMPANY_STATUS_ACTIVE, Jurisdiction.ENGLAND_WALES.toString(), COMPANY_TYPE_PLC, false);
+    }
+
+    @Test
+    void testInvalidCompanyStatus() {
+        validator = factory.getValidator();
+        spec.setJurisdiction(Jurisdiction.ENGLAND_WALES);
+        spec.setCompanyStatus("invalid-company-status");
+        Set<ConstraintViolation<CompanySpec>> violations = validator.validate(spec);
+        assertFalse(violations.isEmpty(), "Expected violations for invalid company status");
+        for (ConstraintViolation<CompanySpec> violation : violations) {
+            assertEquals("Invalid company status", violation.getMessage());
```

Comment:
can you just get the first entry out of this - if this did actually loop to would fail?

Reply:
added assertion directly instead of loop